src/sagemaker_containers/_env.py

```diff
@@ -550,6 +551,15 @@ def job_name(self):  # type: () -> str
         """
         return self._job_name
 
+    @property
+    def parameter_server_num(self):  # type: () -> int
```

Comment:
is this useful for more than just TF?

Reply:
I think so. It's a possible use case for mxnet or pytorch, right? Chainer uses mpi no parameter servers there.

Reply:
MXNet is going to stay out of the user's way as much as possible, so I don't think this would be useful for the current MXNet plan.

For PyTorch, I don't see anything in the [training file](https://github.com/aws/sagemaker-pytorch-container/blob/master/src/sagemaker_pytorch_container/training.py) that would suggest being able to use this. It also does very little in terms of setup.

Reply:
sagemaker-containers filters out all the 'sagemaker_' prefixed hyperparameters. If I do this in sagemaker-tensorflow-containers I have to use a regular hyperparameter and remove it from the dict before calling the user_module. It's really hacky.

Reply:
fair enough

Reply:
@icywang86rui You can use the constructor to do the following (https://github.com/aws/sagemaker-containers/blob/c31296cc1b0f53dc0879f6d45d328c3d04d004e2/src/sagemaker_containers/beta/framework/__init__.py#L32):

```python

hps = env.read_hyperparameters()
env = create_training_env(hyperparameters=hps)

my_field = hps['field']
```
What can we do to improve this experience in a generic way?

Reply:
can we change this so that we just capture all sagemaker_ parameters in a dict, store the dict in TrainingEnv in addition to the other named attributes? That lets us flexibly pass framework-specific stuff through this package without adding any framework-specific code here.

all sagemaker_ parameters are all captured in a dict and the

Reply:
This is an option @jesterhazy . Add the time, we decided to not do that because the TrainingEnv was designed to be used by customer as well: providing env.sagemaker_hyperparameters could be confusing. We can review the decision.

An alternative of this option if subclassing: class TFEnv(TrainingEnv) in the sagemaker_tensorflow library with the helper functions.

Reply:
My original thought was this can be used as in other framework as well. I don't want to use the normal hyperparameter because that's for customer to use in the training script. If we are only going to do this for Tensorflow, it might make sense to use a different prefix and capture those and store in TrainingEnv like @jesterhazy mentioned.

I will make the change after our discussion this afternoon.

Reply:
I like the idea to add this to all frameworks. Are you planning on making the changes in PyTorch and Chainer?